src/benchmarks/micro/corefx/System.Text.Json/Serializer/ReadJson.Dictionary.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public class ReadDictionary
+    {
+        private const string _jsonString = @"{""Hello"":""World"",""Hello2"":""World2""}";
+
+        [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+        [Benchmark]
+        public Dictionary<string, string> DeserializeDict()
```

Comment:
Maybe there is a way to have a generic benchmark method with the type of the dictionary being a type parameter. All these tests are quite similar.

@adamsitnik, any suggestions on how we could set that up (or do you think the tests are fine as is)?

Reply:
@layomia have you considered adding the Dictionary and ImmutableDictionary to existing JSON serializer benchmarks?

You would need to add the type here:

https://github.com/dotnet/performance/blob/2e05afce5332d5e0a308e6208112b4205d3faf75/src/benchmarks/micro/corefx/System.Text.Json/Serializer/WriteJson.cs#L13-L17

and implement the data generation in:
https://github.com/dotnet/performance/blob/master/src/benchmarks/micro/Serializers/DataGenerator.cs

If you do that you can quickly add the deserialization benchmarks to https://github.com/dotnet/performance/blob/master/src/benchmarks/micro/corefx/System.Text.Json/Serializer/ReadJson.cs